tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,182 @@
+
+snippet:EmptyHandlerAsync
+
+You can implement `IHandleMessages<T>` for multiple different message types within the same class, if it makes sense to logically group the handlers together. Just don't expect any state in class variables to be persisted. NServiceBus will create a new instance of the class for every message that it handles.
+
+snippet:MultiHandler
+
+When NServiceBus starts up, it will find all of these message handler classes and automatically wire them up, so that they will be invoked when messages arrive. There's no special setup or configuration needed.
+
+
+## Exercise
+
+Now let's take the "Hello world" solution we started in the last lesson and modify it to send a message. You can also grab the [completed solution from the last lesson](https://github.com/Particular/docs.particular.net/tree/academy-nsb101/tutorials/nservicebus-101/lesson-1/solution/) to use as a starting point.
+
+When we're done, the ClientUI endpoint will be sending a PlaceOrder message to itself, and then processing that message, as depicted in the following diagram:
+
+![Exercise 2 Diagram](diagram.png)
+
+
+### Create a messages assembly
+
+To share message between endpoints they need to be self-contained within a separate assembly, so let's create that assembly now.
+
+ 1. In the solution, create a new project and select the **Class Library** project type.
+ 1. Set the name of the project to **Messages**.
+ 1. Remove the automatically created **Class1.cs** file from the project. We would prefer to have our messages a little more organized than files in the root of the project. (More on this a little later.)
+ 1. Add the NServiceBus NuGet package to the Messages project.
+ 1. In the **ClientUI** project, add a reference to the **Messages** project.
+
+
+### Create a message
+
+In order to better organize messages, we'll create our first command in a folder called **Commands**.
+
+ 1. In the **Messages** project, create a new folder called **Commands**.
+ 1. In the **Commands** folder, add a new class named `PlaceOrder`.
+ 1. Mark `PlaceOrder` as `public` and implement `ICommand`.
+ 1. Add a public property of type `string` named `OrderId`.
+
+NOTE: The .NET Framework contains another interface named `ICommand` in the `System.Windows.Input` namespace. Be sure that if you use tooling to resolve the namespace, you select `NServiceBus.ICommand`. Most of the types you will need will reside in the `NServiceBus` namespace.
+
+When complete, your `PlaceOrder` class should look like the following:
+
+snippet:PlaceOrder
+
+
+### Create a handler
+
+Now that we've defined a message, we can create a corresponding message handler. Rather than create a new endpoint, let's start by just handling the message locally within the **ClientUI** endpoint.
```

Comment:
> Rather than create a new endpoint, let's start by just handling the message locally within the **ClientUI** endpoint.

I'd skip that now and explain in the later lesson/section